node/src/components/linear_chain.rs

```diff
@@ -169,6 +193,30 @@ impl<I> LinearChain<I> {
         }
         (block, effects)
     }
+
+    /// Adds finality signature to the collection of pending finality signatures.
+    fn add_finality_signature(&mut self, fs: FinalitySignature) {
+        let FinalitySignature {
+            block_hash,
+            public_key,
+            ..
+        } = fs;
+        debug!(%block_hash, %public_key, "received new finality signature");
```

Comment:
Maybe clarify that we don't have the block yet:
```suggestion
        debug!(%block_hash, %public_key, "received new finality signature for unknown block");
```

I'd also call the method `add_pending_finality_signature`.

Reply:
Done in [`c0bd13f` (#683)](https://github.com/CasperLabs/casper-node/pull/683/commits/c0bd13f70e41208382667535631dd62c9a381c5c)